SU2_CFD/include/solvers/CSolver.hpp

```diff
@@ -437,6 +437,40 @@ class CSolver {
                                            unsigned short iMesh,
                                            unsigned short RunTime_EqSystem) { }
 
+  /*! "Add" residual at (iPoint,iVar) to residual variables local to the thread.
+   *  \param[in] iPoint - Point index.
+   *  \param[in] iVar - Variable index.
+   *  \param[in] res - Residual at (iPoint,iVar), e.g. LinSysRes(iPoint,iVar)
+   *  \param[in,out] resRMS - increases by pow(Residual, 2)
+   *  \param[in,out] resMax - increases to max(resMax, Residual)
+   *  \param[in,out] idxMax - changes when resMax increases
+   */
+  static inline void ResidualReductions_PerThread(unsigned long iPoint, unsigned short iVar, su2double res, su2double* resRMS, su2double* resMax, unsigned long* idxMax) {
```

Comment:
Excellent idea. Make sure these two functions are in the protected part of the declarations please.

Reply:
I moved the two new functions below `protected:` in d20b2c5, and some additional residual-related functions in 
296e757.